orangecontrib/geo/widgets/owgeotransform.py

```diff
@@ -40,107 +56,106 @@ class Outputs:
     attr_lon = settings.ContextSetting(None)
     from_idx = settings.Setting("Slovenia 1996 / Slovene National Grid")
     to_idx = settings.Setting("WGS 84")
+    replace_original = settings.Setting(True)
 
     want_main_area = False
     resizing_enabled = False
 
     EPSG_CODES = get_projections()
 
     class Error(OWWidget.Error):
-        no_lat_lon_vars = Msg(
-                "Data has no latitude and longitude variables.")
+        no_lat_lon_vars = Msg("Data has no latitude and longitude variables.")
 
     def __init__(self):
         super().__init__()
         self.data = None
 
-        box = gui.vBox(self.controlArea, box="Coordinates:")
-
+        layout = QFormLayout()
+        gui.widgetBox(self.controlArea, box="Coordinates:", orientation=layout)
         self.variable_model = DomainModel(
             order=DomainModel.MIXED, valid_types=(ContinuousVariable,))
-
-        geo_args = dict(contentsLength=100, searchable=True,
-                        model=self.variable_model, orientation=Qt.Horizontal)
-
-        gui.comboBox(box, self, "attr_lat", label="Latitude:", **geo_args)
-        gui.comboBox(box, self, "attr_lon", label="Longitude:", **geo_args)
-
-        box = gui.vBox(self.controlArea, box="Transformation:")
-
-        transform_model = PyListModelTooltip()
-        transform_model.setParent(self)
-        transform_model[:] = chain(list(self.EPSG_CODES))
-        transform_model.tooltips[:] = chain(list(self.EPSG_CODES))
-
-        proj_args = dict(contentsLength=100, searchable=True,
-                         model=transform_model, orientation=Qt.Horizontal,
-                         sendSelectedValue=True)
-
-        gui.comboBox(box, self, "from_idx", label="From:", **proj_args)
-        gui.comboBox(box, self, "to_idx", label="To:", **proj_args)
+        args = dict(contentsLength=100, searchable=True,
+                    model=self.variable_model, orientation=Qt.Horizontal)
+        layout.addRow("Latitude:", gui.comboBox(None, self, "attr_lat", **args))
+        layout.addRow("Longitude:", gui.comboBox(None, self, "attr_lon", **args))
+        layout.addWidget(
+            gui.checkBox(
+                None, self, "replace_original", "Replace original coordinates",
+                tooltip="If unchecked, the original coordinates are retained "
+                        "and new coordinates are added as separate variables."))
+
+        layout = QFormLayout()
+        gui.widgetBox(self.controlArea, "Transformation:", orientation=layout)
+        args["model"] = PyListModel(self.EPSG_CODES)
```

Comment:
I think tooltips are redundant if they are the same as options. Or have I overlooked something.

At any rate, `chain` was redundant, and `list` as well, because `PyListModel` (or `[:]`) accepts iterable, so dict keys are OK.

-----

I reused args from before, to keep the basic settings the same for all combos.

------

`sendSelectedValue` is ignored when using `model`.